mod/cli/pkg/components/logger.go

```diff
@@ -33,15 +33,17 @@ import (
 
 type LoggerInput struct {
 	depinject.In
-	Out     io.Writer
 	AppOpts servertypes.AppOptions
+	Out     io.Writer
 }
 
 // ProvideLogger creates a the default phuslu logger.
 // It reads the log level and format from the server context.
 func ProvideLogger(
 	in LoggerInput,
-) (log.Logger, error) {
+) log.Logger {
```

Comment:
try to use our log.Logger[any] wherever you can

Reply:
not really sure why, but depinject is unable to resolve whenever i try to provide our `AdvancedLogger[any, sdk.Logger]` to the cosmos components...

despite our AdvancedLogger adhering to the same interface after applying generics.